xarray/backends/netCDF4_.py

```diff
@@ -182,7 +182,10 @@ def _extract_nc4_variable_encoding(variable, raise_on_invalid=False,
 def _open_netcdf4_group(filename, mode, group=None, **kwargs):
     import netCDF4 as nc4
 
-    ds = nc4.Dataset(filename, mode=mode, **kwargs)
+    if isinstance(filename, nc4.Dataset):
+        ds = filename
+    else:
+        ds = nc4.Dataset(filename, mode=mode, **kwargs)
 
     with close_on_error(ds):
```

Comment:
We don't want to automatically close already open netCDF4 datasets if there's a problem loading a group.